lang/test/org/partiql/lang/syntax/SqlParserTestBase.kt

```diff
@@ -227,17 +247,20 @@ abstract class SqlParserTestBase : TestBase() {
     protected fun checkInputThrowingParserException(
         input: String,
         errorCode: ErrorCode,
-        expectErrorContextValues: Map<Property, Any>
+        expectErrorContextValues: Map<Property, Any>,
+        targetParsers: Set<ParserTypes> = defaultParserTypes
     ) {
 
         softAssert {
-            try {
-                parser.parseAstStatement(input)
-                fail("Expected ParserException but there was no Exception")
-            } catch (pex: ParserException) {
-                checkErrorAndErrorContext(errorCode, pex, expectErrorContextValues)
-            } catch (ex: Exception) {
-                fail("Expected ParserException but a different exception was thrown \n\t  $ex")
+            targetParsers.forEach {
+                try {
+                    it.parser.parseAstStatement(input)
```

Comment:
Considering introducing the new builder, do we need to change this?

Reply:
No, because we directly use the internal implementations in the tests. Then, we can call parseAstStatement directly on the implemented `Parser`s.